common/scripted_effects/wc_sacrifice_scripted_effects.txt

```diff
@@ -138,18 +138,115 @@ caltulate_PREV_offmap_old_gods_currency_insanity_effect = {
 				}
 				else_if = {
 					limit = { tier = KING }
-					PREV = { change_variable = { which = local_received_offmap_old_gods_currency_insanity_variable value = 300 } }
+					PREV = { change_variable = { which = local_received_offmap_old_gods_currency_insanity_variable value = 350 } }
 				}
 				else = {
 					limit = { tier = EMPEROR }
-					PREV = { change_variable = { which = local_received_offmap_old_gods_currency_insanity_variable value = 500 } }
+					PREV = { change_variable = { which = local_received_offmap_old_gods_currency_insanity_variable value = 650 } }
 				}
 			}
-			# Child
+			#Class
 			if = {
-				limit = { is_adult = no }
+				limit = {
+					OR = {
+						is_1_2_magic_class_trigger = yes
+						is_1_2_physical_class_trigger = yes
+					}
+				}
+				PREV = { change_variable = { which = local_received_offmap_old_gods_currency_insanity_variable value = 50 } }
+			}
+			if = {
+				limit = {
+					OR = {
+						is_3_4_magic_class_trigger = yes
+						is_3_4_physical_class_trigger = yes
+					}
+				}
 				PREV = { change_variable = { which = local_received_offmap_old_gods_currency_insanity_variable value = 100 } }
 			}
+			if = {
+				limit = {
+					OR = {
+						is_5_6_magic_class_trigger = yes
+						is_5_6_physical_class_trigger = yes
+					}
+				}
+				PREV = { change_variable = { which = local_received_offmap_old_gods_currency_insanity_variable value = 150 } }
+			}
+			if = {
+				limit = {
+					OR = {
+						is_7_8_magic_class_trigger = yes
+						is_7_8_physical_class_trigger = yes
+					}
+				}
+				PREV = { change_variable = { which = local_received_offmap_old_gods_currency_insanity_variable value = 250 } }
+			}
+			if = {
+				limit = {
+					OR = {
+						is_9_10_magic_class_trigger = yes
+						is_9_10_physical_class_trigger = yes
+					}
+				}
+				PREV = { change_variable = { which = local_received_offmap_old_gods_currency_insanity_variable value = 400 } }
+			}
+			#Races that cant have class
+			if = {
+				limit = {
+					is_whelp_trigger = yes
+				}
+				PREV = { change_variable = { which = local_received_offmap_old_gods_currency_insanity_variable value = 50 } }
+			}
+			if = {
+				limit = {
+					is_wyrm_trigger = yes
+				}
+				PREV = { change_variable = { which = local_received_offmap_old_gods_currency_insanity_variable value = 100 } }
+			}
+			if = {
+				limit = {
+					is_drake_trigger = yes
+				}
+				PREV = { change_variable = { which = local_received_offmap_old_gods_currency_insanity_variable value = 150 } }
+			}
+			if = {
+				limit = {
+					is_dragon_trigger = yes
+				}
+				PREV = { change_variable = { which = local_received_offmap_old_gods_currency_insanity_variable value = 200 } }
```

Comment:
I still don't support it, but there's a bug. The wyrms are the biggest dragons, they shouldn't give less than the drakes.
The scaling is wrong. `creature_red_wyrm` is like 2 times better than `creature_red_dragon`, `creature_red_dragon` is 2 times better than `creature_red_drake`.

Reply:
A legendary druid gives more than a wyrm without an aspect (so Neltharion, Sindragosa, etc.)

Reply:
so 200 and 400 for the last two?